caches.go

```diff
@@ -92,6 +93,36 @@ func (rcc *clientRegionCache) clientDown(c hrpc.RegionClient) map[hrpc.RegionInf
 	return downregions
 }
 
+// Collects information about the clientRegion cache and stores them in two maps to reduce
```

Comment:
nit: the code could look more organized if all the new functions appear together at the end of the file